src/game/behaviors/fish.inc.c

```diff
@@ -1,163 +1,233 @@
+    if (o->oDistanceToMario < o->oFishGroupUnk104 + 150.0f) {
         o->oAction = 2;
+    }
 }
 
-void ActionFishGroup2(void) {
-    f32 sp1C = o->oPosY - gMarioObject->oPosY;
-    s32 sp18;
+void bhv_fish_group_2(void) {
+    f32 fishY = o->oPosY - gMarioObject->oPosY;
+    s32 distance;
     o->oFishGroupUnkF8 = gMarioObject->oPosY + o->oFishGroupUnkFC;
     if (o->oTimer == 0) {
         o->oFishGroupUnk110 = RandomFloat() * 300.0f;
         o->oFishGroupUnk100 = RandomFloat() * 1024.0f + 1024.0f;
         o->oFishGroupUnk108 = RandomFloat() * 4.0f + 8.0f + 5.0f;
-        if (o->oDistanceToMario < 600.0f)
-            sp18 = 1;
-        else
-            sp18 = (s32)(1.0 / (o->oDistanceToMario / 600.0));
-        sp18 *= 127;
-        PlaySound2(SOUND_GENERAL_MOVING_WATER);
-    }
-    if (o->oTimer < LEVEL_SA)
+        if (o->oDistanceToMario < 600.0f) {
+            distance = 1;
+        }
+        else {
+            distance = (s32)(1.0 / (o->oDistanceToMario / 600.0));
+        }
+            distance *= 127;
+            PlaySound2(SOUND_GENERAL_MOVING_WATER);
+    }
+    if (o->oTimer < LEVEL_SA) {
         func_8029ED98(0, 4.0f);
-    else
+    }
+    else {
         func_8029ED98(0, 1.0f);
-    if (o->oForwardVel < o->oFishGroupUnk108)
+    }
+    if (o->oForwardVel < o->oFishGroupUnk108) {
         o->oForwardVel = o->oForwardVel + 0.5;
+    }
     o->oFishGroupUnkF8 = gMarioObject->oPosY + o->oFishGroupUnkFC;
+    // Rotate fish towards mario
     obj_rotate_yaw_toward(o->oAngleToMario + 0x8000, o->oFishGroupUnk100);
+    // If fish groups are too close regroup
     if (o->oPosY < o->oFishGroupUnkF4 - 50.0f) {
-        if (sp1C < 0.0f)
-            sp1C = 0.0f - sp1C;
-        if (sp1C < 500.0f)
-            func_802BF59C(2);
-        else
-            func_802BF59C(4);
+        if (fishY < 0.0f) {
+            fishY = 0.0f - fishY;
+        }
+        if (fishY < 500.0f) {
+            regroup_fish(2);
+        }
+        else {
+            regroup_fish(4);
+        }
     } else {
         o->oPosY = o->oFishGroupUnkF4 - 50.0f;
-        if (sp1C > 300.0f)
+        if (fishY > 300.0f) {
             o->oPosY -= 1.0f;
+        }
     }
-    if (o->oDistanceToMario > o->oFishGroupUnk110 + 500.0f)
+    if (o->oDistanceToMario > o->oFishGroupUnk110 + 500.0f) {
         o->oAction = 1;
+    }
 }
-
-void ActionFishGroup0(void) {
+// Randomly move/animate/scale fish
+void bhv_fish_group_0(void) {
     func_8029ED98(0, 1.0f);
     o->header.gfx.unk38.animFrame = (s16)(RandomFloat() * 28.0f);
     o->oFishGroupUnk10C = RandomFloat() * 300.0f;
     obj_scale(RandomFloat() * 0.4 + 0.8);
     o->oAction = 1;
 }
 
-void (*sFishGroupActions[])(void) = { ActionFishGroup0, ActionFishGroup1, ActionFishGroup2 };
+// Array of methods
+void (*sFishGroupActions[])(void) = { bhv_fish_group_0, bhv_fish_group_1, bhv_fish_group_2 };
 
+// In SA alter fish pathing if they hit a wall.
 void bhv_fish_group_2_loop(void) // TODO rename
```

Comment:
Finishing file documentation should handle all TODO's, if thats the goal of this PR.